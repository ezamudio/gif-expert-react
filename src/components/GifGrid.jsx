import { PropTypes } from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifCard } from './GifCard';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {/* Esto se podr√≠a hacer con un custom component */}
      {isLoading && <h2>Cargando...</h2>}
      <div className='card-grid'>
        {images.map((image) => (
          <GifCard key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
