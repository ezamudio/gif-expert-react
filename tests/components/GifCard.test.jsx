import { render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components/GifCard';

describe('Pruebas en <GifCard/>', () => {
  const title = 'Prueba 1';
  const url = 'https://picsum.photos/200';

  test('Debe de hacer match con el Snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifCard title={title} url={url} />);
    //screen.debug();
    /* expect(screen.getByRole('img').src).toBe(url);
    expect(screen.getByRole('img').alt).toBe(title); */
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GifCard title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
