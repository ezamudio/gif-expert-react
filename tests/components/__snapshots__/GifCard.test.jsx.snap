// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard/> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba 1"
      src="https://picsum.photos/200"
    />
    <p>
      Prueba 1
    </p>
  </div>
</div>
`;
